parcels/particleset.py

```diff
@@ -324,9 +324,12 @@ def execute(self, pyfunc=AdvectionRK4, endtime=None, runtime=None, dt=1.,
                 time = max(next_prelease, next_input, next_output, next_movie, endtime)
             self.kernel.execute(self, endtime=time, dt=dt, recovery=recovery, output_file=output_file)
             if abs(time-next_prelease) < tol:
-                self.add(ParticleSet(fieldset=self.fieldset, time=time, lon=self.repeatlon,
-                                     lat=self.repeatlat, depth=self.repeatdepth,
-                                     pclass=self.repeatpclass))
+                pset_new = ParticleSet(fieldset=self.fieldset, time=time, lon=self.repeatlon,
+                                       lat=self.repeatlat, depth=self.repeatdepth,
+                                       pclass=self.repeatpclass)
+                for p in pset_new:
+                    p.dt = dt
+                self.add(pset_new)
```

Comment:
Why not have `dt` as an argument to `ParticleSet.__init__`, which is then directly passed to `Particle.__init__`? Then, this could simply be
```
self.add(ParticleSet(fieldset=self.fieldset, time=time, lon=self.repeatlon,
                                   lat=self.repeatlat, depth=self.repeatdepth,
                                   pclass=self.repeatpclass, dt=dt))
```
Would that not be cleaner?

Reply:
If dt is an argument of ParticleSet constructor, there is no point on having a dt argument at pset.execute(), because they would conflict.